hail/src/main/scala/is/hail/utils/RestartableByteArrayInputStream.scala

```diff
@@ -0,0 +1,69 @@
+package is.hail.utils
+
+import java.io.{ IOException, InputStream }
+
+// not thread safe
+class RestartableByteArrayInputStream (
+  start: Int,
+  private[this] var end: Int,
+  private[this] var buf: Array[Byte]
```

Comment:
Minor nit: I think it would be a bit cleaner to structure the constructors:
```scala
class RestartableByteArrayInputStream extends InputStream {
  var off: Int = 0
  private var end: Int = 0
  private var buf: Array[Byte] = null
  // no requires in constructor, only in restart
  def this(start: Int, end: Int, buf: Array[Byte]) {
	this()
    restart(buf, start, end)
  }
  ...
}
```

Reply:
This is indeed nicer, done.